alacritty_terminal/src/term/mod.rs

```diff
@@ -320,9 +498,65 @@ impl<T> Term<T> {
             selection: None,
             cell_width: size.cell_width as usize,
             cell_height: size.cell_height as usize,
+            is_fully_damaged: true,
+            damage,
         }
     }
 
+    #[must_use]
+    pub fn damage(&mut self, selection: Option<SelectionRange>) -> TermDamage<'_> {
+        // Ensure that we damage entire terminal in insert mode.
+        if self.mode.contains(TermMode::INSERT) {
+            self.is_fully_damaged = true;
+        }
+
+        // Add information about old cursor position and new one if they are not the same, so we
+        // cover everything that was produced by `Term::input`.
+        if self.damage.old_cursor != self.grid.cursor.point {
+            self.damage.damage_point(self.damage.old_cursor);
+        }
```

Comment:
What if the line changed? What if the cursor jumped back and forth since the last frame?

I don't see what benefit this provides beyond `damage_cursor` already done right below.

Reply:
The point is to cover `Term::input`, if you won't do that you won't damage content written by it properly. As an example look at the test using `input`, that's exactly why damaging old cursor is required, since you expand from it up to new one.